Hearthstone/app/src/main/java/com/example/hearthstone/ui/cardOverview/CardOverviewFragment.kt

```diff
@@ -5,21 +5,91 @@ import androidx.fragment.app.Fragment
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
+import androidx.core.text.HtmlCompat
 import androidx.databinding.DataBindingUtil
+import androidx.fragment.app.viewModels
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.navArgs
 import com.example.hearthstone.R
+import com.example.hearthstone.data.model.HSCardsByClassModel
 import com.example.hearthstone.databinding.FragmentCardOverviewBinding
+import dagger.hilt.android.AndroidEntryPoint
 
-
-class CardOverviewFragment : Fragment() {
+@AndroidEntryPoint
+class CardOverviewFragment() : Fragment() {
     private lateinit var binding: FragmentCardOverviewBinding
+    private val viewModel: CardOverviewViewModel by viewModels()
+    private val args: CardOverviewFragmentArgs by navArgs()
+    private lateinit var hsCard: HSCardsByClassModel
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
         // Inflate the layout for this fragment
-        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_card_overview, container, false)
+        binding =
+            DataBindingUtil.inflate(inflater, R.layout.fragment_card_overview, container, false)
+        viewModel.getAllCards()
+        checkFavorite()
+        goSearchPage()
+
+        viewModel.fav.observe(viewLifecycleOwner) {
+            binding.icon.isChecked = it
+        }
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        args.hsCard?.let {
+            hsCard = it.copy(
+                text = it.text?.let { text ->
+                    HtmlCompat.fromHtml(
```

Comment:
Why are you using HTML?

Reply:
In the api, when obtaining the cards there is a property called "text", it's the effect of the card, but some cards have a secret effect and it comes like this: <b>Secret:</b>When your opponent.... so that the labels wouldn't be seen in the app, Rafa recommended me to use that

Reply:
Got it.